### ISSUE TITLE
Error LatLng2 error

### ISSUE THREAD

**Issue by theglamolondon** [fleaflet/flutter_map#888]
The argument type 'LatLng (where LatLng is defined in C:\SDK\flutter_new\.pub-cache\hosted\pub.dartlang.org\latlong2-0.8.0\lib\latlong\LatLng.dart)' can't be assigned to the parameter type 'LatLng (where LatLng is defined in C:\SDK\flutter_new\.pub-cache\hosted\pub.dartlang.org\latlong-0.6.1\lib\latlong\LatLng.dart)'.

pubspec.yaml content :
...
  flutter_map: ^0.12.0
  latlong2: ^0.8.0

**Comment by ibrierley** [comment]
I think you want to remove latlong2 from pubspec maybe (as I think its included in flutter_map), and just add import 'package:latlong2/latlong.dart'; at the top of your code.

**Comment by aodbee** [comment]
use this
import 'package:latlong/latlong.dart';
It Work for me

**Comment by ibcoleman** [comment]
@theglamolondon  I feel your pain. From what I can tell (a lot of Issues have been closed as Duplicates, pointing to other Issues, etc., etc... but still not a whole lot clarity) the latest flutter_map (0.12.0) still depends on the "orphaned" latlong (0.6.1) rather than latlong2. Maybe the next release will cut latlong loose.

**Comment by ibrierley** [comment]
Afaik, you don't need any LatLng in your pubspec.
You may need import 'package:latlong2/latlong.dart'; in your package that references any points that are needed to access flutter_map

Maybe someone else can clear this up though, as it is a source of confusion, and should be made very clear in big bold letters on the readme :D

**Comment by ibcoleman** [comment]
``@ibrierley: Maybe I'm running into problems because I make explicit reference to LatLng to set the center when I construct a new Map.

If I just try to import 'package:latllong2/latlong.dart' I get a broken build because it can't resolve that package:

```
Error: Could not resolve the package 'latlong2' in 'package:latlong2/latlong.dart'.
lib/client/tva_asset_service.dart:6:8: Error: Not found: 'package:latlong2/latlong.dart'
import 'package:latlong2/latlong.dart';
```

If I go ahead and replace `latlong: 0.6.0` with `latlong2: 0.8.0` in pubspec.yaml, then update the imports throughout the code, I get an error like:

```
lib/ui/asset/asset_detail.dart:202:21: Error: The argument type 'LatLng/*1*/' can't be assigned to the parameter type 'LatLng/*2*/'.
 - 'LatLng/*1*/' is from 'package:latlong2/latlong.dart' ('../../../.pub-cache/hosted/pub.dartlang.org/latlong2-0.8.0/lib/latlong.dart').
 - 'LatLng/*2*/' is from 'package:latlong/latlong.dart' ('../../../.pub-cache/hosted/pub.dartlang.org/latlong-0.6.1/lib/latlong.dart').
            center: _getCenter(context.watch<AssetAggregate>())
```

**Comment by ibrierley** [comment]
Might be worth doing a "flutter pub deps" in your terminal to see what dependencies and versions you are using. I'm not an expert on this stuff, so I'm hoping someone else may be able to help more. It looks a bit like it's still trying to load latlong(not 2) 0.6.1 from your cache.

**Comment by ibcoleman** [comment]
Hmm. Thanks for the pub deps tip, I hadn't tried that, but it looks like `flutter_map:0.12.0` is still pointing to `latlong:0.6.1` (are the people successfully using latlong2 linking 0.12.0 or are they running the latest build from the github repo?):

```
|-- flutter_map 0.12.0
|   |-- async...
|   |-- collection...
|   |-- flutter...
|   |-- flutter_image 3.0.0
|   |   '-- flutter...
|   |-- latlong 0.6.1
|   |   |-- console_log_handler 1.1.6
|   |   |   |-- ansicolor 1.1.1
|   |   |   |-- intl...
|   |   |   '-- logging...
|   |   |-- intl...
|   |   |-- logging...
|   |   '-- validate 1.7.0
|   |-- meta...
|   |-- positioned_tap_detector 1.0.3
|   |   '-- flutter...
|   |-- proj4dart 1.0.5
|   |   |-- meta...
|   |   |-- mgrs_dart 1.0.1
|   |   |   '-- unicode 0.2.4
|   |   |       '-- lists 0.1.6
|   |   '-- wkt_parser 1.0.7
|   |-- transparent_image 1.0.0
|   |-- tuple 1.0.3
|   |   '-- quiver 2.1.5
|   |       |-- matcher...
|   |       '-- meta...
|   '-- vector_math...
```

**Comment by ibrierley** [comment]
I'm pointing mine at the git repo.
Mine says flutter_map 0.12 also, but differently shows latlong2 0.8.0.
I note in flutter_maps pubspec it specifies ^0.8.0

**Comment by ibcoleman** [comment]
Ah, okay, that makes sense. I'm still linking to the pub.dev dep. Sounds like they've got the same version number but that the repo master branch is ahead(?). Thanks so much for helping me figure out what's going on, though. Felt like I was going crazy!  😄

**Comment by kengu** [comment]
That's correct @ibcoleman - master is usually ahead of current published version

**Comment by gairick-saha** [comment]
Thankx this works

**Comment by jarrodmedrano** [comment]
> I'm pointing mine at the git repo.
> Mine says flutter_map 0.12 also, but differently shows latlong2 0.8.0.
> I note in flutter_maps pubspec it specifies ^0.8.0

Can you post your code how you are linking to the git version?

**Comment by ibrierley** [comment]
There's no need to link to the Git version now, as the master has been updated to nullsafety which uses latlong2...

  flutter_map: any
  latlong2: ^0.8.0

**Comment by ibrierley** [comment]
Just to add, latlong has always been a bit of a pain with flutter_map, so if you get any problems, do a "flutter pub deps" and paste what it shows.

**Comment by gairick-saha** [comment]
[latlong2migrated.zip](https://github.com/fleaflet/flutter_map/files/6718936/latlong2migrated.zip)

I have fixed this issue by modifying the source code a little bit of some packages that I have installed on my system.
I found this packages depends on latlong 0.6.1 so latlong 0.6.1 gets automatically downloaded into "flutter/.pub-cache" which cause some issue importing the proper packages. Properly check which package is conflicting with latlong 0.6.1 & try removing conflicting packages along with latlong 0.6.1 from "flutter/.pub-cache".
